packages/contractkit/src/utils/phone-number-lookup/phone-number-lookup.ts

```diff
@@ -97,27 +98,39 @@ export async function postToPhoneNumPrivacyService<ResponseType>(
   }
 
   const { pgpnpUrl } = context
-  const res = await fetch(pgpnpUrl + endpoint, {
-    method: 'POST',
-    headers: {
-      Accept: 'application/json',
-      'Content-Type': 'application/json',
-      Authorization: authHeader,
-    },
-    body: bodyString,
-  })
-
-  if (!res.ok) {
+
+  const tries = 3
+  let lastError
+  for (let i = 1; i <= tries; i++) {
```

Comment:
There's a retry method in our utils library that we should use:
https://github.com/celo-org/celo-monorepo/blob/master/packages/utils/src/async.ts#L35
More code re-use and it has a configurable back-off. This prevents our server from getting pounded with traffic whenever there's a drop in availability (which can sometimes make the problem worse).

Reply:
Right I looked at that. That method won't allow us to automatically throw on certain error codes. Is that okay or should I implement a new helper?